### PR TITLE
Add enforce_sub_prefix option to AFQDataset and load_afq_data

### LINKED ISSUE #94: Match subject identifiers with and without the "sub-" prefix
### Problem

When using `load_afq_data` or `AFQDataset`, if the one of `subjects.csv` or `nodes.csv` has a "sub-" prefix for the subject identifiers and the other does not, then it won't find the matching phenotypic info for each subject.

### Proposed solution

Add a `drop_sub_prefix=False` parameter to `load_afq_data` and `AFQDataset`. When `False` it should reproduce the current behavior. When `True` it should drop the "sub-" prefix from both the nodes and subject dataframes before merging.

### PR DESCRIPTION
**Pull request by richford**
Resolves #94

### CODE REVIEW

**Comment by richford** [comment]
I think this is ready for review @arokem, whenever you get a chance.

File: afqinsight/datasets.py [hunk 1]
```diff
@@ -154,6 +171,12 @@ def load_afq_data(
         If True, return diffusion metrics averaged along the length of each
         bundle.
 
+    enforce_sub_prefix : bool, default=True
+        If True, standardize subject IDs to start with the prefix "sub-".
+        This is useful, for example, if the subject IDs in the nodex.csv file
```
Comment (arokem) [inline afqinsight/datasets.py:176]:
```suggestion
        This is useful, for example, if the subject IDs in the nodes.csv file
```

File: afqinsight/datasets.py [hunk 2]
```diff
@@ -382,6 +411,12 @@ class AFQDataset:
         IDs with the session IDs. This is useful when subjects have multiple
         sessions and you with to disambiguate between them.
 
+    enforce_sub_prefix : bool, default=True
+        If True, standardize subject IDs to start with the prefix "sub-".
+        This is useful, for example, if the subject IDs in the nodex.csv file
```
Comment (arokem) [inline afqinsight/datasets.py:416]:
```suggestion
        This is useful, for example, if the subject IDs in the nodes.csv file
```

File: afqinsight/tests/test_datasets.py [hunk 3]
```diff
@@ -285,6 +294,7 @@ def test_load_afq_data(dwi_metrics):
         target_cols=["test_class"],
         label_encode_cols=["test_class"],
         return_bundle_means=True,
+        enforce_sub_prefix=False,
     )
```
Comment (arokem) [inline afqinsight/tests/test_datasets.py:298]: Is there anything to test here? Or are the subject IDs all already prefixed with `sub` here?

**Review by arokem — COMMENTED** [review]
Looks good! I found a couple of typos, and was wondering whether this could be tested beyond the currently implemented smoke test.

File: afqinsight/datasets.py (hunk 1 above)
Comment (richford) [reply afqinsight/datasets.py]: Thanks. Resolved!

File: afqinsight/datasets.py (hunk 2 above)
Comment (richford) [reply afqinsight/datasets.py]: Thanks! Resolved

File: afqinsight/tests/test_datasets.py (hunk 3 above)
Comment (richford) [reply afqinsight/tests/test_datasets.py:342]: I added a test for this in [6be7a0d](https://github.com/richford/AFQ-Insight/pull/96/commits/6be7a0d86e553850d54e19cee21dcb080f92ae95)

**Comment by richford** [comment]
I'm going to recommend merging and eating the -0.005% decrease in coverage since the code changes are in `datasets.py` and the coverage for that file is 100%

**Pull request merged**